feat: first set of mapping implementation #213

BaptisteCecconi opened the pull request:
First implementation for get_id(), get_name(), get_description() and get_url().

clnsmth commented:
Thanks for the PR @BaptisteCecconi!

I have a few comments that I'll post in the code review.

clnsmth commented on src/soso/strategies/spase.py:
```diff
@@ -41,22 +41,42 @@ def __init__(self, file: str, **kwargs: dict):
+        self.namespaces = {"spase": "http://www.spase-group.org/data/schema"}
         self.kwargs = kwargs
 
     def get_id(self) -> None:
```
Update the return type from `None` to `str` to match the result. If the return value can be `None`, because the SPASE element is not required, then define the return type as `Union[str, None]`

clnsmth commented on src/soso/strategies/spase.py:
```diff
@@ -41,22 +41,42 @@ def __init__(self, file: str, **kwargs: dict):
+        )
         return delete_null_values(dataset_id)
 
     def get_name(self) -> None:
```
Update the return type from `None` to `str` to match the result. If the return value can be `None`, because the SPASE element is not required, then define the return type as `Union[str, None]`.

clnsmth commented on src/soso/strategies/spase.py:
```diff
@@ -41,22 +41,42 @@ def __init__(self, file: str, **kwargs: dict):
+        )
         return delete_null_values(name)
 
     def get_description(self) -> None:
```
Update the return type from `None` to `str` to match the result. If the return value can be `None`, because the SPASE element is not required, then define the return type as `Union[str, None]`.

clnsmth commented on src/soso/strategies/spase.py:
```diff
@@ -41,22 +41,42 @@ def __init__(self, file: str, **kwargs: dict):
+        )
         return delete_null_values(description)
 
     def get_url(self) -> None:
```
Update the return type from `None` to `str` to match the result. If the return value can be `None`, because the SPASE element is not required, then define the return type as `Union[str, None]`.

clnsmth reviewed (commented):
For each property conversion method (e.g. `soso.strategies.spase.get_id`), ensure the `@pytest.mark.skipif` decorator is removed from the corresponding test in `soso.tests.strategies`, so that the test runs.

It looks like these can be removed:

https://github.com/clnsmth/soso/blob/de3b4307146207d96085408d3e322a92efd6b2dd/tests/test_strategies.py#L55

https://github.com/clnsmth/soso/blob/de3b4307146207d96085408d3e322a92efd6b2dd/tests/test_strategies.py#L68

https://github.com/clnsmth/soso/blob/de3b4307146207d96085408d3e322a92efd6b2dd/tests/test_strategies.py#L80

BaptisteCecconi commented:
Ok, I updated the return types and removed the `@pytest.mark.skipif` decorators when applicable.

I still have a question on `tests/test_main.py::test_convert_returns_expected_properties`. It is failing if I don't skip it, and I don't see how to fix this one.
